Assets/GothicVR/Scripts/World/SunMover.cs

```diff
@@ -0,0 +1,61 @@
+using GVR.Demo;
+using GVR.Util;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SunMover : MonoBehaviour
+{
+    private DebugSettings.SunMovementPerformance sunPerformanceSetting;
+    public GameObject sun;
+
+    public DateTimeGameEvent secondChangeEvent;
+    public DateTimeGameEvent minuteChangeEvent;
+    public DateTimeGameEvent hourChangeEvent;
+
+    private void OnEnable()
+    {
+
+    }
+    private void OnDisable()
+    {
+        switch (sunPerformanceSetting)
+        {
+            case DebugSettings.SunMovementPerformance.EveryIngameSecond:
+                secondChangeEvent.OnEventRaised -= RotateSun;
+                break;
+            case DebugSettings.SunMovementPerformance.EveryIngameMinute:
+                minuteChangeEvent.OnEventRaised -= RotateSun;
+                break;
+            case DebugSettings.SunMovementPerformance.EveryIngameHour:
+                hourChangeEvent.OnEventRaised -= RotateSun;
+                break;
+            default:
+                break;
+        }
+    }
+    private void Start()
+    {
+        sunPerformanceSetting = SingletonBehaviour<DebugSettings>.GetOrCreate().SunMovementPerformanceValue;
+        switch (sunPerformanceSetting)
+        {
+            case DebugSettings.SunMovementPerformance.EveryIngameSecond:
+                secondChangeEvent.OnEventRaised += RotateSun;
+                break;
+            case DebugSettings.SunMovementPerformance.EveryIngameMinute:
+                minuteChangeEvent.OnEventRaised += RotateSun;
+                break;
+            case DebugSettings.SunMovementPerformance.EveryIngameHour:
+                hourChangeEvent.OnEventRaised += RotateSun;
+                break;
+            default:
+                break;
+        }
+    }
+    void RotateSun(DateTime time)
+    {
+        var xRotation = 270f + (15f * (time.Hour + (time.Minute / 60f) + (time.Second / 3600f)));
```

Comment:
I remember this line... :-D
Please re-add the extensive comment I created in my old version. Otherwise reviewing this line will be a pain in a month or later.